BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.h

```diff
@@ -0,0 +1,163 @@
+//
+//  BOXURLSessionCacheClient.h
+//  BoxContentSDK
+//
+//  Created by Thuy Nguyen on 3/3/17.
+//  Copyright © 2017 Box. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@interface BOXURLSessionTaskCachedInfo : NSObject
+//@see comments above BOXURLSessionCacheClient for detailed explaination of those properties
+
+@property (nonatomic, strong, readwrite) NSString *backgroundSessionId;
+@property (nonatomic, assign, readwrite) NSUInteger sessionTaskId;
+@property (nonatomic, strong, readwrite) NSString *destinationFilePath;
+@property (nonatomic, strong, readwrite) NSData *resumeData;
+@property (nonatomic, strong, readwrite) NSURLResponse *response;
+@property (nonatomic, strong, readwrite) NSData *responseData;
+@property (nonatomic, strong, readwrite) NSError *error;
+
+@end
+
+@protocol BOXURLSessionCacheClientDelegate <NSObject>
+
+@optional
+// allow delegate to encrypt data before BOXURLSessionCacheClient persists it to disk
+- (NSData *)encryptData:(NSData *)data;
```

Comment:
As with delegate patterns per Apple's recommendations, we should include the object in the delegate method, `- (NSData *)sessionCacheClient:(BOXURLSessionCacheClient *)client willEncryptData:(NSData *)data`.